### PR TITLE
 Signal/Interrupt package

### PR DESCRIPTION
**Pull request by cfromknecht**
Moves most of the contents in `signal.go` into it's own package named `signal`. This is done in preparation of having standalone watchtowers, so each daemon can share this code.  

Notable changes:
 - handlers are now executed in LIFO order. Before we were registering closures that executed in order, though this allows us to have similar semantics as `defer`. The handlers in `lnd.go` have now been separated, and are registered after each `Start()`.
 - handlers are now `func() error` instead of `func()`. All of the handlers we use are of this type, so it makes it cleaner even though we don't handle the errors during an interrupt.

### CODE REVIEW

**Comment by cfromknecht** [comment]
One question I have is whether we even need `AddInterruptHandler`, instead of just preferring to `defer` the `Stop()` methods. If we also defer the log message stating that shutdown is complete at the beginning of `lndMain`, everything should already be in lifo order.

**Comment by cfromknecht** [comment]
I added two commits that impl the `defer` approach, lmk which you all prefer

File: signal/signal.go [hunk 1]
```diff
@@ -12,33 +12,30 @@ import (
 
 var (
 	// interruptChannel is used to receive SIGINT (Ctrl+C) signals.
-	interruptChannel chan os.Signal
+	interruptChannel = make(chan os.Signal, 1)
```
Comment (Roasbeef) [inline signal/signal.go]: Nice addition! Should've been this way all along.

**Comment by cfromknecht** [comment]
Rationale for using defer: the current system executes FIFO order, which is fine since only two functions are registered (one of which is composite, and for the most part, executes `Stop` in LIFO order of resource creation). Instead of reimplementing a LIFO queue, I chose defer since it's built into the language, is consistent with all other teardown paradigms in lnd, and results in less overall code.

**Comment by cfromknecht** [comment]
can squash the last two commits if rationale is sound

**Comment by Roasbeef** [comment]
👍

**Comment by cfromknecht** [comment]
squash commits and rebased on master

**Pull request merged**